File: 1/05.py
import re
def n_gram(seq, n):
  n_gram = []
  for i in range(len(seq)-n+1):
    n_gram.append(seq[i:i+n])
  return n_gram
s = "I am an NLPer"
words = re.split(r'\s+', s)
word_n_gram = n_gram(words, 2)
char_n_gram = n_gram(s, 2)
print(word_n_gram)
print(char_n_gram)

File: 1/05.js
function n_gram(seq, n) {
  var result = []
  for (var i=0; i<seq.length-n+1; ++i) {
    result.push(seq.slice(i, i+n))
  }
  return result
}
var s = "I am an NLPer"
var words = s.split(/\s+/)
var word_n_gram = n_gram(words, 2)
var chars = s.split('')
var char_n_gram = n_gram(chars, 2)
console.log(word_n_gram)
console.log(char_n_gram)
